xdsl/interpreters/experimental/wgsl_printer.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import annotations
+
+from functools import singledispatchmethod
+from typing import IO, cast
+
+from xdsl.dialects import arith, builtin, gpu, memref
+from xdsl.dialects.memref import MemRefType
+from xdsl.ir import Operation, SSAValue
+from xdsl.ir.core import Attribute
+
+
+class WGSLPrinter:
+    name_dict: dict[SSAValue, str] = dict()
+    count = 0
+
+    def wgsl_name(self, v: SSAValue):
+        if v not in self.name_dict.keys():
+            if v.name_hint is not None:
```

Comment:
I think this won't work if you have two ssa values with the same name hint? You also need to keep track of the name hints you accepted (you can see the normal printer for that. Might be something worth factoring out later?)

Reply:
It is a hazard at this point, but we'll improve that later with proper test cases! The first use plan is to just parse-and-translate, and the parser handles that case either way.
Just trying to get a first part in to improve the different aspects later !
NB: This is clearly one aspect that I would love to codeshare with the internal printer, which already handles this, right?